### PR TITLE
Create ajax error with server response text

### LINKED ISSUE #10091: Return server response as a text instead of statusText
## Motivation
Right now, when a vector tile fails to load, Mapbox is returning an error with  the statusText which in most cases is empty.

Here is a link to the piece of code I would like to change
https://github.com/mapbox/mapbox-gl-js/blob/94e4714c0ab2896304d72e6322b30fe0bad392a5/src/util/ajax.js#L230
We can change it to
```js
 callback(new AJAXError(xhr.responseText || xhr.statusText, xhr.status, requestParameters.url)); 
```

### LINKED ISSUE #10097: How to do Heatmap with lines ?
Hi, 

I can't find how to do how to do an heatmap with lines similar to 
image

I have a Geojson source with a lot of routes and I would like to style with a scale of color where high traffic road are "hot" and low traffic are "cold"

Thank you very much

### PR DESCRIPTION
**Pull request by kikonejacob**
Closes #10091

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: src/util/ajax.js [hunk 1]
```diff
@@ -151,7 +151,12 @@ function makeFetchRequest(requestParameters: RequestParameters, callback: Respon
                 return finishRequest(response, cacheableResponse, requestTime);
 
             } else {
-                return callback(new AJAXError(response.statusText, response.status, requestParameters.url));
+                response.text().then(text => {
```
Comment (asheemmamoowala) [inline src/util/ajax.js:154]: Would you consider attempting to read `response.text()` only when `statusText` is empty?

**Review by asheemmamoowala — CHANGES_REQUESTED** [review]
@kikonejacob Thank you for the change. For consistency, I think this change should be made in `makeXMLHttpRequest` as well. 

Could you also add some unit tests in `https://github.com/mapbox/mapbox-gl-js/blob/main/test/unit/util/ajax.test.js` with mocked responses.

**Comment by ryanhamley** [comment]
Closing as stale. @kikonejacob if you want to pick this back up, feel free to re-open
